### ISSUE TITLE
Adding to the model dialogue - 2 - new keyboards

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#5126]
From the prediction package, it copes with many more models than we have now, - or will have in the near future - in our "usual" dialogues.  I strongly suggest that adding more keyboards to the "halfway" Model > Model dialogue could be both interesting and also important.  

We need to make a selection (of course), but much of relatively recent research has enabled a very wide collection of models to be fitted is a roughly consistent way.  So having then for a range of situations in a single dialogue could be really helpful.

I describe the more complete picture more in the next issue - where we add a prediction dialogue - also important.  Here is my initial list of additional keyboards - mainly working backwards from prediction.

I am here sticking to the idea of (mainly) having a keyboard per package, with the name of the package dictating the keyboard.  But we may relax that later, particularly as there are some packages with just a single function.

I am also trying to consider the range of tasks where it could be good to fit a model.  So we already have  lme4 so we can fit multi-level models.  Fitting models for categorical data is an obvious need.  I like to be able to fit models for rates and proportions (betareg) and so on.  In my choices below I am also influenced by the number of downloads - using popularity of the package as an indication (not always reliable) of the value of the package. For example for the [ordinal package](https://www.rdocumentation.org/packages/ordinal/versions/2018.8-25) we see it was last updated in August 2018 and is on the 98th percentile with 262 downloads per month.

This may split into a number of issues, because each keyboard is likely to add a major component.  We are also into a subject that might appeal greatly to our Colombo partners?

**1. survival keyboard.**  We are thinking of a survival menu, but perhaps we could start here with a survival keyboard.  Perhaps @lilyclements could advise on this?  coxph and survreg are already mentioned with the prediction, so those are a start.
**2. survey keyboard** A number of the svy models are specifically mentioned, e.g. svyglm.  There are quite a lot.  We will (later) want "proper" dialogues for them, but who knows when, and it would still be sensible (as with survival) to also have them here.
**3. betareg** This is for rates and proportions, because they often assume a beta model.  There is particularly the betareg function, and also betamix and betatree.
**4. ordinal** Looks good.  I wonder if this might also be the same keyboard as for nnet, which includes multinom.
5. ** pscl** This is particularly for discrete numerical data, and includes hurdle models where zero is special (e.g. zeroinfl function).
6. **kernlab** This is on the 99% point with 650 downloads per month and seems popular for machine learning - and also with the "usual" prediction.
7. To be continued

**Comment by dannyparsons** [comment]
@lilyclements do these functions require a survival object as the input? If so that would need some thought as currently we're assuming a formula as the main specification.

This might be the same for the `survey` package where there is a `survey` object that many of the functions use.

**Comment by rdstern** [comment]
I agree that items that need a special type of object are a new step and currently we are not that far in the functions in the keyboards.  I think there are quite a lot of survival "keys" that can be used.  Then the next step in this dialogue (or maybe it is another dialogue, because it is a sort of Use Model idea, is to be able to start with special objects.  This is for surveys, survival, extremes and so on.  Important, but there is a lot to do before that.    I would like to discuss this topic when you are back - with you, David and Lily.

**Comment by lilyclements** [comment]
@dannyparsons most require a survival object as the input:
* `survreg` "usually" requires the `Surv` object. I don't know of a case where it doesn't, but I know it does not have to
* `cox.zph` has its input as a `coxph` model as this is testing the assumptions for the model. This is more on the test side than the modelling side so not sure how appropriate it is here anyway.
* `clogit` doesn't use the `Surv` object as the response
